server/channels/app/migrations.go

```diff
@@ -582,6 +596,118 @@ func (s *Server) doPostPriorityConfigDefaultTrueMigration() error {
+	for _, property := range existingProperties {
+		existingPropertiesMap[property.Name] = property
+	}
+
+	expectedPropertiesMap := map[string]*model.PropertyField{
+		contentFlaggingPropertyNameFlaggedPostId: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameFlaggedPostId,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameStatus: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameStatus,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameReportingUserID: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameReportingUserID,
+			Type:    model.PropertyFieldTypeUser,
+		},
+		contentFlaggingPropertyNameReportingReason: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameReportingReason,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameReportingComment: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameReportingComment,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameReportingTime: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameReportingTime,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameReviewerUserID: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameReviewerUserID,
+			Type:    model.PropertyFieldTypeUser,
+		},
+		contentFlaggingPropertyNameActorUserID: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameActorUserID,
+			Type:    model.PropertyFieldTypeUser,
+		},
+		contentFlaggingPropertyNameActorComment: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameActorComment,
+			Type:    model.PropertyFieldTypeText,
+		},
+		contentFlaggingPropertyNameActionTime: {
+			GroupID: group.ID,
+			Name:    contentFlaggingPropertyNameActionTime,
+			Type:    model.PropertyFieldTypeText,
+		},
+	}
+
+	var propertiesToUpdate []*model.PropertyField
+	var propertiesToCreate []*model.PropertyField
+
+	for name, expectedProperty := range expectedPropertiesMap {
+		if _, exists := existingPropertiesMap[name]; exists {
+			property := existingPropertiesMap[name]
+			property.Type = expectedProperty.Type
+			propertiesToUpdate = append(propertiesToUpdate, property)
+		} else {
+			propertiesToCreate = append(propertiesToCreate, expectedProperty)
+		}
+	}
+
+	for _, property := range propertiesToCreate {
+		if _, err := s.propertyService.CreatePropertyField(property); err != nil {
+			return fmt.Errorf("failed to create content flagging property: %q, error: %w", property.Name, err)
+		}
+	}
+
+	if _, err := s.propertyService.UpdatePropertyFields(group.ID, propertiesToUpdate); err != nil {
+		return fmt.Errorf("failed to update content flagging property fields: %w", err)
+	}
+
+	if err := s.Store().System().Save(&model.System{Name: contentFlaggingSetupDoneKey, Value: "true"}); err != nil {
+		return fmt.Errorf("failed to save content flagging setup done flag in system store %w", err)
+	}
```

Comment:
Why don't we hold version for the value here? With that you could actually check the version and then determine if there is a migration required. Getting all the values from get-go seems a bit inefficient to me. WDYT?

```json
"content_flagging_setup_done": "v1"
"content_flagging_setup_done": "v2"
```